src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -426,5 +432,929 @@ public void EmptyRecord()
+  IL_0026:  stfld      ""int C.<Y>k__BackingField""
+  IL_002b:  call       ""void System.Console.WriteLine(object)""
+  IL_0030:  ret
+}");
+        }
+
+        [Fact]
+        public void WithExpr15()
+        {
+            var src = @"
+data class C(int X, int Y)
+{
+    public C Clone() => null;
+    public static void Main()
+    {
+        var c = new C(0, 0);
+        c = c with { = 5 };
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (8,22): error CS1525: Invalid expression term '='
+                //         c = c with { = 5 };
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "=").WithArguments("=").WithLocation(8, 22)
+            );
+        }
+
+        [Fact]
+        public void WithExpr16()
+        {
+            var src = @"
+data class C(int X, int Y)
+{
+    public C Clone() => null;
+    public static void Main()
+    {
+        var c = new C(0, 0);
+        c = c with { X = };
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (8,26): error CS1525: Invalid expression term '}'
+                //         c = c with { X = };
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "}").WithArguments("}").WithLocation(8, 26)
+            );
+        }
+
+        [Fact]
+        public void WithExpr17()
+        {
+            var src = @"
+class B
+{
+    public int X { get; }
+    private B Clone() => null;
+}
+data class C(int X) : B
+{
+    public static void Main()
+    {
+        var b = new B();
+        b = b with { };
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (12,13): error CS8803: The 'with' expression requires the receiver type 'B' to have a single accessible non-inherited instance method named "With".
+                //         b = b with { };
+                Diagnostic(ErrorCode.ERR_NoSingleCloneMethod, "b").WithArguments("B").WithLocation(12, 13)
+            );
+        }
+
+        [Fact]
+        public void WithExpr18()
+        {
+            var src = @"
+class B
+{
+    public int X { get; }
+    protected B Clone() => null;
+}
+data class C(int X) : B
+{
+    public static void Main()
+    {
+        var b = new B();
+        b = b with { };
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void WithExpr19()
+        {
+            var src = @"
+class B
+{
+    public int X { get; }
+    protected B Clone() => null;
+}
+data class C(int X)
+{
+    public static void Main()
+    {
+        var b = new B();
+        b = b with { };
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (12,13): error CS8803: The 'with' expression requires the receiver type 'B' to have a single accessible non-inherited instance method named "With".
```

Comment:
>With [](start = 160, length = 4)

Clone?

Reply:
And a few others

---
In reply to: [413365604](https://github.com/dotnet/roslyn/pull/43249#discussion_r413365604) [](ancestors = 413365604)